chapters/fund/participation.md

```diff
@@ -2,59 +2,102 @@
+  `uint investmentAmount` - The quantity of tokens of the `investmentAsset`
+  `uint requestedShares` - The quantity of fund shares for the request
+  `uint timestamp` - The timestamp of the block containing the subscription transaction
+  `uint atUpdateId` - [CHECK]
+
+
+#### Enums
 
-    `address investmentAsset`
-    `uint investmentAmount`
-    `uint requestedShares`
-    `uint timestamp`
-    `uint atUpdateId`
+None.
 
+#### Modifiers
 
-Enums
 None.
 
 
-Public State Variables
+#### Events
+
+`RequestExecuted()`
+
+    `address investmentAsset` - The address of the asset token with which the subscription is made.
+    `uint investmentAmount` - The quantity of tokens of the `investmentAsset`
+    `uint requestedShares` - The calculated quantity of fund shares created for the subscription
+    `uint timestamp` - The timestamp of the block containing the subscription transaction
+    `uint atUpdateId` - [CHECK]
+
+This event is triggered when a subscription is successfully executed by the fund.
+
+`SuccessfulRedemption()`
+
+    `uint quantity` - The quantity of shares successfully redeemed
+
+This event is triggered when a redemption is successfully executed by the fund.
+
+
+#### Public State Variables
 
 `mapping (address => Request) public requests`
 
-A mapping which assigns an investor's address to a `Request` struct.
+A public mapping which assigns an investor's address to a `Request` struct.
 
+`mapping (address => bool) public investAllowed`
+
+A public mapping which specifies all asset tokens which have been enabled for subscription to the Melon fund.
 
 `uint public SHARES_DECIMALS`
 
-An integer which specifies the number of decimals or divisibility of a single share.
+An integer which specifies the decimal precision of a single share. The value is set to 18.
+
+
+
+#### Public Functions
+
+`function enableInvestment(address[] _assets) public auth`
 
+This function requires that the caller is the `owner` or the current contract. The function iterates over an array of provided asset token addresses, ensuring each is registered with the Melon fund's PriceFeed, and ensures that registered asset token addresses are set to `true` in the `investAllowed` mapping.
 
 
-Public Functions
+`function disableInvestment(address[] _assets) public auth`
 
-`function requestInvestment(uint requestedShares, uint investmentAmount, address investmentAsset) external`
+This function requires that the caller is the `owner` or the current contract. The function iterates over an array of provided asset token addresses and ensures that asset token addresses are set to `false` in the `investAllowed` mapping.
 
-This function ensures that the fund is not shutdown, then populates the `Request` struct and adds this to the `requests` mapping corresponding to the `msg.sender`.  This function also sets `SHARE_DECIMALS` to 18.
+
+`function requestInvestment(uint requestedShares, uint investmentAmount, address investmentAsset) external payable amguPayable`
```

Comment:
Should we mention reasoning behind two step process here?

Reply:
agree opening an issue for this